util.c

```diff
@@ -481,10 +486,35 @@ bool should_bypass_proxy(const char *url, const char *bypass_list) {
         // Copy rule to temporary buffer
         strncat(bypass_rule, rule_start, rule_len);
 
+        // Strip and parse port from bypass rule
+        port = strchr(bypass_rule, ':');
+        int32_t bypass_rule_port = 0;
+        if (port) {
+            *port = 0;
+            bypass_rule_port = atoi(port + 1);
+        }
+
         // If the rule matches hostname of url then bypass proxy
-        if (str_wildcard_match(host, bypass_rule, true))
+        if (str_wildcard_match(host, bypass_rule, true)) {
             should_bypass = true;
 
+            // Check bypass rule port
+            if (bypass_rule_port) {
+                if (!host_port) {
+                    // Infer default host port from url scheme
+                    char *scheme = get_url_scheme(url, "http");
+                    if (scheme) {
+                        host_port = get_scheme_default_port(scheme);
+                        free(scheme);
+                    }
+                }
+
+                // If the host port doesn't match the bypass rule port then don't bypass
+                if (bypass_rule_port != host_port)
+                    should_bypass = false;
+            }
+        }
+
```

Comment:
It does not seem to bypass a proxy if the default port is set explicitly in the request "http://microsoft.com:80/" or "https://microsoft.com:443/" and microsoft.com is set to bypass, and vise versa "http://microsoft.com/" or "https://microsoft.com/" is requested and http://microsoft.com:80 or https://microsoft.com:443 is set to bypass.

Reply:
I have these two unit tests which I believe cover those instances:
```c
    // Bypass due to matching domain with any port
    {"http://microsoft.com:88/", "microsoft.com", true},
    // Bypass due to matching domain with explicit port
    {"http://microsoft.com:88/", "microsoft.com:88", true},
```
First one uses explicitly defined port 88 and it is bypassed because it is microsoft.com.
Second one uses explicitly defined port in both instances and is bypassed.

Reply:
Default ports:
```c
   // Bypass due to matching domain with explicit port
   {"http://microsoft.com/", "microsoft.com:80", true},
   // Bypass due to matching domain with explicit port
   {"https://microsoft.com/", "microsoft.com:443", true},
```

Reply:
It is near equivalent to the one I had:
```c
{"https://microsoft.com/", "*.com:443", true},
```
Nevertheless, I have added those to the test cases.